test/Error/aggregate_error.js

```diff
@@ -0,0 +1,384 @@
+    },
+    {
+        name: "errors iterable to list failures",
+        body: function ()
+        {
+            class TestError extends Error {}
+
+            var case1 = {
+                get [Symbol.iterator]() {
+                    throw new TestError();
+                }
+            };
+            assert.throws(() => {
+                var obj = new AggregateError(case1);
+            }, TestError);
+
+            var case2 = {
+                get [Symbol.iterator]() {
+                    return {};
+                }
+            };
+            assert.throws(() => {
+                var obj = new AggregateError(case2);
+            }, TypeError);
+
+            var case3 = {
+                [Symbol.iterator]() {
+                    throw new TestError();
+                }
+            };
+            assert.throws(() => {
+                var obj = new AggregateError(case3);
+            }, TestError);
+
+            var case4 = {
+                [Symbol.iterator]() {
+                  return "a string";
+                }
+            };
+            assert.throws(() => {
+                var obj = new AggregateError(case4);
+            }, TypeError);
+
+            var case5 = {
+                [Symbol.iterator]() {
+                    return undefined;
+                }
+            };
+            assert.throws(() => {
+                var obj = new AggregateError(case5);
+            }, TypeError);
+
+            var case6 = {
+                [Symbol.iterator]() {
+                    return {
+                        get next() {
+                            throw new TestError();
+                        }
+                    }
+                }
+            };
+            assert.throws(() => {
+                var obj = new AggregateError(case6);
+            }, TestError);
+
+            var case7 = {
+                [Symbol.iterator]() {
+                    return {
+                        get next() {
+                            return {};
+                        }
+                    }
+                }
+            };
+            assert.throws(() => {
+                var obj = new AggregateError(case7);
+            }, TypeError);
+
+            var case8 = {
+                [Symbol.iterator]() {
+                    return {
+                        next() {
+                            throw new TestError();
+                        }
+                    }
+                }
+            };
+            assert.throws(() => {
+                var obj = new AggregateError(case8);
+            }, TestError);
+
+            var case9 = {
+                [Symbol.iterator]() {
+                    return {
+                        next() {
+                            return undefined;
+                        }
+                    }
+                }
+            };
+            assert.throws(() => {
+                var obj = new AggregateError(case9);
+            }, TypeError);

```

Comment:
The next few tests appear to be about iterators that don't provide objects -would be good to specify that.

Reply:
Sorry I'm not sure what‘s the point.

Reply:
Basically that I'd like it to be clearer when re-reading the test file what it's doing/what these cases are for.

It appears from reading them that they're testing behaviour when an iterator doesn't provide an object - but there's no comment description saying that that is what they're doing/why they're here.

Reply:
Thanks for the response. I have tried to add some comments above the case. Does they seems enough?